navext/navext-runtime/src/test/java/com/foursoft/harness/navext/runtime/io/validation/XMLValidationTest.java

```diff
@@ -107,4 +110,28 @@ void validateDoubleHyphenInXmlCommentError() throws IOException {
                 .withMessageContaining("--");
     }
 
+    @Test
+    void testValidateXML() {
```

Comment:
Testing 4 different cases / scenarios in a single test method is very smelly. The test with the noopConsumer is almost useless. It should at least check if the consumer is called when required. Even better that it is called with the correct output.